spring-boot-project/spring-boot-tools/spring-boot-buildpack-platform/src/main/java/org/springframework/boot/buildpack/platform/docker/configuration/DockerConfiguration.java

```diff
@@ -146,16 +151,25 @@ public String getCertificatePath() {
 			return this.certificatePath;
 		}
 
+		public Integer getSocketTimeout() {
+			return this.socketTimeout;
+		}
+
 		public static DockerHostConfiguration forAddress(String address) {
-			return new DockerHostConfiguration(address, null, false, null);
+			return new DockerHostConfiguration(address, null, false, null, null);
+		}
+
+		public static DockerHostConfiguration forAddress(String address, Integer socketTimeout) {
```

Comment:
I think a separate `withSocketTimeout` instance method would be better. It would reduce the scope of the changes as the `for…` methods no longer need to be overloaded.